### DISCUSSION TITLE
Composite Primary Keys in Supabase?

### DISCUSSION THREAD

**Discussion by switchupcb** [supabase/supabase#10585] [Questions]
How to create a composite primary key through the Supabase UI?

_Note: It's possible to implement using a remote editor._

**Comment by GaryAustin1** [comment]
You should be able to just select multiple columns as primary key with their check boxes.

**Reply by switchupcb** [reply]
This works for non-unique composite primary keys. Otherwise, both fields end up unique.

**Reply by GaryAustin1** [reply]
I've not used that, but if you are saying checking the primary box on each also makes NON unique AND you don't want that, then you are probably stuck using SQL.   I don't know of other options.  You might make a feature request, if it does not already exist.

EDIT.  It looks like columns of the composite key cannot be unique in Postgres.  So not sure what you are trying to achieve beyond what the UI does.

**Reply by switchupcb** [reply]
```sql
UNIQUE(a, b)
```

**Reply by GaryAustin1** [reply]
OK.   If you use the UI to set multiple columns to primary key I believe the result is `PRIMARY KEY (a, b)`.  So the pair has to be unique, but not each individual column.  You are showing UNIQUE now, is your question how to do a UNIQUE constraint on multiple columns with the UI (you can't ) or a Composite primary key (I believe that works).
